### PR TITLE
[WAREHOUSE]: add ComponentEntity, residues, test 

### PR DESCRIPTION
**Pull request by artem-tkach**

### CODE REVIEW

File: warehouse/src/main/java/io/skai/warehouse/model/ComponentEntity.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Id
+    @Immutable
+    public static final EntityField<ComponentEntity, Long> ID = INSTANCE.field(ComponentTable.TABLE.id);
+
+    @Required
+    @Immutable
+    public static final EntityField<ComponentEntity, String> NAME = INSTANCE.field(ComponentTable.TABLE.name);
+    //
+    //RESIDUES
```
Comment (asinchuk) [inline warehouse/src/main/java/io/skai/warehouse/model/ComponentEntity.java]: This is unnecessary comment

File: warehouse/src/main/java/io/skai/warehouse/repository/ComponentPersistence.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import io.skai.warehouse.command.CreateComponentCommand;
+import io.skai.warehouse.model.ComponentEntity;
+import jakarta.annotation.Resource;
+
+import java.util.Collection;
+
+public class ComponentPersistence {
+
+    private final PersistenceLayer<ComponentEntity> persistenceLayer;
+    @Resource
```
Comment (asinchuk) [inline warehouse/src/main/java/io/skai/warehouse/repository/ComponentPersistence.java:13]: new line

File: warehouse/src/main/java/io/skai/warehouse/service/impl/ComponentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private final ComponentPersistence componentPersistence;
+    private final ComponentCommandsBuilder componentCommandsBuilder;
+
+    @Override
+    public List<ComponentDto> create(List<ComponentDto> components) {
+        List<CreateComponentCommand> commands = componentCommandsBuilder.buildCreateCommands(components);
+
+        CreateResult<ComponentEntity, Identifier<ComponentEntity>> result = componentPersistence.create(commands);
+
+        return seq(result.getChangeResults())
```
Comment (asinchuk) [inline warehouse/src/main/java/io/skai/warehouse/service/impl/ComponentServiceImpl.java]: Use java stream

File: warehouse/src/main/java/io/skai/warehouse/service/impl/ComponentServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    private ComponentDto mapResults(EntityChangeResult<ComponentEntity, Identifier<ComponentEntity>, CreateEntityCommand<ComponentEntity>> result) {
+        Long id = getValuefromResult(result, ComponentEntity.ID);
+        String name = getValuefromResult(result, ComponentEntity.NAME);
+        Double count = getValuefromResult(result, ComponentEntity.COUNT);
+        Double reserved = getValuefromResult(result, ComponentEntity.RESERVED);
+
+        return new ComponentDto(id, name,count, reserved);
+    }
+
+    private <T> T getValuefromResult(EntityChangeResult<ComponentEntity, Identifier<ComponentEntity>, CreateEntityCommand<ComponentEntity>> result,
```
Comment (asinchuk) [inline warehouse/src/main/java/io/skai/warehouse/service/impl/ComponentServiceImpl.java]: This is utility method, please move it to separate class.

Comment (asinchuk) [inline warehouse/src/main/java/io/skai/warehouse/service/impl/ComponentServiceImpl.java]: bad name, it should not duplicate parameters.

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (asinchuk) [inline warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java:21]: ?

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+@Testcontainers
+@SpringBootTest
+class ComponentServiceImplTest {
+
+    private static final String SCREEN = "screen";
+    private static final String KEYBOARD = "keyboard";
+    private static final String TOUCHPAD = "touchpad";
+    @Container
```
Comment (asinchuk) [inline warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java:31]: ?

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        registry.add("spring.datasource.username", container::getUsername);
+        registry.add("spring.datasource.password", container::getPassword);
+    }
+
+    @Test
+    void shouldWriteToDbThenReturnIt() {
+        List<ComponentDto> inputData = getInputData();
+        List<ComponentDto> result = componentService.create(inputData);
+        assertThat(result).hasSize(3)
+                .allMatch(n -> n.id() > 0)
```
Comment (asinchuk) [inline warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java]: what n stands for?

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                .extracting("name")
+                .containsExactlyInAnyOrder(TOUCHPAD, KEYBOARD, SCREEN);
```
Comment (asinchuk) [inline warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java:73]: if first check fail, this wouldn't be verified, every validation in test should be atomic.

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        registry.add("spring.datasource.password", container::getPassword);
+    }
+
+    @Test
+    void shouldWriteToDbThenReturnIt() {
+        List<ComponentDto> inputData = getInputData();
+        List<ComponentDto> result = componentService.create(inputData);
+        assertThat(result).hasSize(3)
+                .allMatch(n -> n.id() > 0)
+                .extracting("name")
```
Comment (asinchuk) [inline warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java]: please move to well named constant

File: warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java (hunk 8 above)
Comment (artem-tkach) [reply warehouse/src/test/java/io/skai/warehouse/service/impl/ComponentServiceImplTest.java:73]: @asinchuk 
if i write it separately like this
`assertThat(result).hasSize(3)
                .allMatch(componentDto -> componentDto.id() > 0);
        assertThat(result)
                .extracting("name")
                .containsExactlyInAnyOrder(TOUCHPAD, KEYBOARD, SCREEN);` 
part with "extracting" will be not checked too if id checks fill fail. I wan't to check condition : ID were generated(>0) AND name are same as id input parameters so i used one check.

**Review by asinchuk — APPROVED** [review]

**Pull request merged**